### PR TITLE
Added support for RoBERTa Language Model

### PR DESCRIPTION
**Pull request by franzmpub**
# Pull Request      

## What does this PR do?

Adds support for RoBERTa Language Model, following the same pattern as used for other LMs (e.g. XLMR).       
 
## Description

Tested for training/indexing/search.

### Request Review

Be sure to **request a review** from one or more reviewers (unless the PR is to an unprotected branch).

## Versioning

When opening a PR to make changes to PrimeQA (i.e. `primeqa/`) master, be sure to increment the version following
[semantic versioning](https://semver.org/).  The VERSION is stored [here](https://github.com/primeqa/primeqa/blob/main/VERSION) 
and is incremented using `bump2version {patch,minor,major}` as described in the (development guide documentation)[development.html].

 - [x] Have you updated the VERSION?
 - [ ] Or does this PR not change the `primeqa` package or was not into master?

After pulling in changes from master to an existing PR, ensure the VERSION is updated appropriately.
This may require bumping the version again if it has been previously bumped.

If you're not quite ready yet to post a PR for review, feel free to open a [draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

### After Merging
If merging into master and VERSION was updated, after this PR is merged:
- [ ] [Create a release](https://docs.github.com/en/github/administering-a-repository/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release) from the master with version equal to [VERSION](https://github.com/primeqa/primeqa/blob/main/VERSION)
- [ ] Not merging to master or VERSION not updated

### Checklist

Review the following and mark as completed:
- [ ] [Tag an issue or issues](#what-does-this-pr-do) this PR addresses.
- [x] Added [description](#description) of changes proposed.
- [x] Review requested as [appropriate](#request-review).
- [x] Version bumped as [appropriate](#Versioning).
- [ ] New classes, methods, and functions documented.
- [ ] Documentation for modified code is updated.
- [ ] Built documentation to confirm it renders as expected (see [here](development.md)).
- [ ] Code cleaned up and commented out code removed.
- [ ] Tests added to ensure all functionalities tested at >= 60% unit test coverage (see [here](development.md)).
- [ ] Code cleaned up and commented out code removed.
- [ ] Release created [as needed](#after-merging) after merging.

### CODE REVIEW

File: primeqa/ir/dense/colbert_top/colbert/modeling/hf_colbert_roberta.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import torch.nn as nn
+
+from transformers import AutoTokenizer, RobertaModel
+
+from primeqa.ir.dense.colbert_top.colbert.utils.utils import torch_load_dnn
+from primeqa.ir.dense.colbert_top.colbert.utils.utils import print_message
+
+class HF_ColBERT_Roberta(RobertaModel):
+#class HF_ColBERT_XLMR(PreTrainedModel):
```
Comment (seirasto) [inline primeqa/ir/dense/colbert_top/colbert/modeling/hf_colbert_roberta.py:9]: remove comment

File: primeqa/ir/dense/colbert_top/colbert/modeling/tokenization/doc_tokenization_roberta.py
Comment (seirasto) [inline primeqa/ir/dense/colbert_top/colbert/modeling/tokenization/doc_tokenization_roberta.py:1]: Can the comments be cleaned up here as well?

File: primeqa/ir/dense/colbert_top/colbert/modeling/tokenization/query_tokenization_roberta.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        self.tok = HF_ColBERT_Roberta.raw_tokenizer_from_pretrained(model_type)
+        self.query_maxlen = query_maxlen
+
+        self.Q_marker_token, self.Q_marker_token_id = '[Q]', self.tok.convert_tokens_to_ids('madeupword0000')
+        self.mask_token, self.mask_token_id = self.tok.pad_token, self.tok.pad_token_id
+
+#        assert self.Q_marker_token_id == 1 and self.mask_token_id == 103
+        self.used = False
+
+    # tokenizer is not used colbert code base, but is implemented in QueryTokenizer
```
Comment (seirasto) [inline primeqa/ir/dense/colbert_top/colbert/modeling/tokenization/query_tokenization_roberta.py:32]:
```suggestion
    # tokenizer is not used in the colbert code base, but is implemented in QueryTokenizer
```

File: primeqa/ir/dense/colbert_top/colbert/modeling/tokenization/query_tokenization_roberta.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    def tokenize(self, batch_text, add_special_tokens=False):
+        raise NotImplementedError()
+
+    # encode is not used colbert code base, but is implemented in QueryTokenizer
```
Comment (seirasto) [inline primeqa/ir/dense/colbert_top/colbert/modeling/tokenization/query_tokenization_roberta.py:36]:
```suggestion
    # encode is not used in the colbert code base, but is implemented in QueryTokenizer
```

**Review by seirasto — APPROVED** [review]
LGTM. There are a couple of files where I believe commented out code can be cleaned up and a few minor suggestions.

File: primeqa/ir/dense/colbert_top/colbert/modeling/hf_colbert_roberta.py (hunk 1 above)
Comment (franzmpub) [reply primeqa/ir/dense/colbert_top/colbert/modeling/hf_colbert_roberta.py]: Addressed in an update

File: primeqa/ir/dense/colbert_top/colbert/modeling/tokenization/doc_tokenization_roberta.py
Comment (franzmpub) [reply primeqa/ir/dense/colbert_top/colbert/modeling/tokenization/doc_tokenization_roberta.py:1]: Addressed in an update

**Review by bhavani105 — APPROVED** [review]
LGTM

Please add testcase
Approving

File: primeqa/ir/dense/colbert_top/colbert/modeling/tokenization/query_tokenization_roberta.py
Comment (seirasto) [inline primeqa/ir/dense/colbert_top/colbert/modeling/tokenization/query_tokenization_roberta.py:1]: This file still has a lot of commented out code

**Review by seirasto — APPROVED** [review]
LGTM (still one file with commented out code)

**Review by hrojek — APPROVED** [review]
LGTM

**Pull request merged**